src/fldeff_dig.c

```diff
@@ -0,0 +1,49 @@
+#include "global.h"
+#include "field_effect.h"
+#include "field_player_avatar.h"
+#include "fldeff.h"
+#include "item_use.h"
+#include "overworld.h"
+#include "party_menu.h"
+#include "sprite.h"
+#include "constants/map_objects.h"
+
+static void hm2_dig(void);
+static void sub_80C9AFC(void);
+
+bool8 SetUpFieldMove_Dig(void)
+{
+    if (CanUseEscapeRopeOnCurrMap() == TRUE)
+    {
+        gFieldCallback2 = FieldCallback_PrepareFadeInFromMenu;
+        gPostMenuFieldCallback = hm2_dig;
+        return TRUE;
+    }
+    return FALSE;
+}
+
+static void hm2_dig(void)
```

Comment:
Weird name, consider changing

Reply:
Actually all this kind of name has counterpart in pokeem. Should I still document them with my opinion?

Reply:
Whatever you decide, but the `hm2_` names simply will not fly.